src/contracts/core/DelegationManager.sol

```diff
@@ -37,17 +40,21 @@ contract DelegationManager is
     // @dev Chain ID at the time of contract deployment
     uint256 internal immutable ORIGINAL_CHAIN_ID;
 
-    // @dev Maximum Value for `stakerOptOutWindowBlocks`. Approximately equivalent to 6 months in blocks.
-    uint256 public constant MAX_STAKER_OPT_OUT_WINDOW_BLOCKS = (180 days) / 12;
+    /// @dev The minimum number of blocks to complete a withdrawal of a strategy. 50400 * 12 seconds = 1 week
+    uint256 public constant LEGACY_MIN_WITHDRAWAL_DELAY_BLOCKS = 50_400;
+
+    /// @dev Wed Jan 01 2025 17:00:00 GMT+0000, timestamp used to check whether a pending withdrawal
+    /// should be processed as legacy M2 or with slashing considered.
+    uint32 public constant LEGACY_WITHDRAWALS_TIMESTAMP = 1_735_750_800;
```

Comment:
how's this work, exactly? should this actually be like, Jan 01 2023 or something? 
I'm assuming we check if the value is *greater* than this and then do special logic? if so it seems correct to have this value be before the first withdrawals, to make sure that all existing withdrawals are treated with this legacy logic.

Reply:
having read on, this now makes sense to me -- the key is that historically (i.e. in "M2") withdrawal storage used block numbers, which are currently sitting around 21 million, which is way, way lower than this.
So the check implemented is to check that if a number is less than this constant, then it is a legacy withdrawal using this block-number-based storage. 
This number could actually be much lower (and it may make sense to lower it for testing), and describing it as a "timestamp" is likely not the best naming, but all should work as intended assuming the upgrade occurs after this timestamp.
I'd recommend changing this to something like `LEGACY_WITHDRAWAL_CHECK_VALUE = 1000000000` (as a UTC timestamp this is Sunday, September 9, 2001), and explaining in the comments that it is used to differentiate legacy block number based-storage (less than the value) from timestamps (greater than the value)

Reply:
Updated and implemented your suggestion, LEGACY_WITHDRAWAL_CHECK_VALUE = 1_000_000_000